nfts/src/tests/extrinsics.rs

```diff
@@ -9,375 +9,665 @@ fn origin(account: u64) -> mock::Origin {
+			// Create NFT with new serie id while there is no series already registered
+			assert_ok!(NFTs::create(
+				alice.clone(),
+				data.ipfs_reference.clone(),
+				Some(data.series_id.clone()),
+			));
+			let nft_id = NFTs::nft_id_generator() - 1;
+
+			// Events checks
+			assert_eq!(
+				System::events().last().unwrap().event,
+				Event::NFTs(NFTsEvent::NFTCreated {
+					nft_id,
+					owner: data.owner,
+					series_id: data.series_id.clone(),
+					ipfs_reference,
+					mint_fee: NFTs::nft_mint_fee(),
+				})
+			);
+
+			// Final state checks
+			assert_eq!(NFTs::series_id_generator(), 0);
+			assert_eq!(nft_id, 0);
+			assert_eq!(NFTs::series(&data.series_id), Some(NFTSeriesDetails::new(ALICE, true)));
+			assert_eq!(NFTs::data(0), Some(data.clone()));
+			assert_eq!(Balances::free_balance(ALICE), alice_balance - NFTs::nft_mint_fee());
 		})
 	}
-}
 
-mod burn {
-	use super::*;
+	#[test]
+	fn create_ok_associated_with_existing_serie() {
+		ExtBuilder::default().caps(vec![(ALICE, 1000)]).build().execute_with(|| {
+			// State checks
+			assert_eq!(NFTs::nft_id_generator(), 0);
+			assert_eq!(NFTs::series_id_generator(), 0);
+
+			// Initial state
+			let alice: mock::Origin = origin(ALICE);
+			let serie_id = vec![50];
+			let ipfs_1_reference = vec![1];
+			let ipfs_2_reference = vec![2];
+			let data_1 = NFTData::new_default(ALICE, ipfs_1_reference.clone(), serie_id.clone());
+			let data_2 = NFTData::new_default(ALICE, ipfs_2_reference.clone(), serie_id.clone());
+			let alice_balance = Balances::free_balance(ALICE);
+
+			// Create NFT with new serie id while there is no series already registered
+			assert_ok!(NFTs::create(
+				alice.clone(),
+				data_1.ipfs_reference.clone(),
+				Some(data_1.series_id.clone()),
+			));
+			let nft_1_id = NFTs::nft_id_generator() - 1;
+
+			// NFT Id check
+			assert_eq!(nft_1_id, 0);
+
+			// Events checks
+			assert_eq!(
+				System::events().last().unwrap().event,
+				Event::NFTs(NFTsEvent::NFTCreated {
+					nft_id: nft_1_id,
+					owner: data_1.owner,
+					series_id: data_1.series_id.clone(),
+					ipfs_reference: ipfs_1_reference,
+					mint_fee: NFTs::nft_mint_fee(),
+				})
+			);
+
+			// Create NFT associated with existing serie
+			assert_ok!(NFTs::create(
+				alice.clone(),
+				data_2.ipfs_reference.clone(),
+				Some(data_2.series_id.clone()),
+			));
+			let nft_2_id = NFTs::nft_id_generator() - 1;
+
+			// NFT Id check
+			assert_eq!(nft_2_id, 1);
+
+			// Events checks
+			assert_eq!(
+				System::events().last().unwrap().event,
+				Event::NFTs(NFTsEvent::NFTCreated {
+					nft_id: nft_2_id,
+					owner: data_2.owner,
+					series_id: data_2.series_id.clone(),
+					ipfs_reference: ipfs_2_reference,
+					mint_fee: NFTs::nft_mint_fee(),
+				})
+			);
+
+			// Final state checks
+			assert_eq!(NFTs::series_id_generator(), 0);
+			assert_eq!(NFTs::series(&data_1.series_id), Some(NFTSeriesDetails::new(ALICE, true)));
```

Comment:
Move the instantiation of "NFTSeriesDetails::new(ALICE, true)" to a variable. Do the same thing for the line below